assets/dev/js/editor/components/template-library/manager.js

```diff
@@ -10,14 +10,52 @@ TemplateLibraryManager = function() {
 		errorDialog,
 		layout,
 		templatesCollection,
-		defaultScreen,
+		defaultRoute = 'library/templates/pages',
 		config = {},
+		modalConfig = {},
 		screens = {},
-		startIntent = {},
 		filterTerms = {};
 
 	const initLayout = function() {
 		layout = new TemplateLibraryLayoutView( { pages: screens } );
+
+		layout.getModal().on( 'hide', function() {
```

Comment:
Add a method to close the route directly, something like:
elementorCommon.route.close( 'library' );

Reply:
Not possible because it will no affect external closing like on click outside or `esc` key.